### PR TITLE
Fix cu_hxr energy gain profile calculation

### PR DESCRIPTION
**Pull request by williamColocho**
A few bugs introduced while reworking the code.

### CODE REVIEW

File: lcls_tools/common/data/model_general_calcs.py
Comment (nneveu) [inline lcls_tools/common/data/model_general_calcs.py:1]: This file is already on this repo with these functions? Maybe this wasn't meant to be updated?

**Review by nneveu — COMMENTED** [review]
I'm not sure I see what changed in the `model_general_calcs.py` file? Maybe an accidental addition to this commit/PR?

**Comment by nneveu** [comment]
Also, let me know if you want help resolving the conflicts.

**Comment by williamColocho** [comment]
I would like to help resolving the conflicts.  I don't fully understand being ahead and behind.

What should my regular workflow be to stay on the main branch?

William
________________________________

**Comment by robnagler** [comment]
> I don't fully understand being ahead and behind.

Ahead means that the branch has changes that are not yet added to main, i.e. the changes you desire to add to main in the PR.

Behind means that the main branch has changes that have yet to be merged into your branch, i.e. other PRs being merged into main since you started development (or the last time you merged main).

> What should my regular workflow be to stay on the main branch?

Merge main into your working branch every day before beginning work and possibly multiple times in a day, depending on the state of the branch. Depending on the complexity of changes, you may be able to rebase (apply your changes on top of any new changes in main) instead of merging. This avoids interleaved changes when the PR is created. 

Personally, for larger projects, I always merge and resolve conflicts, and then I recreate a new branch with a squash merge just before the PR. This way there's a single commit with a cohesive description for the PR, which can help reviewers. My branches tend to be very messy, because I commit multiple times in a day to backup my work.

**Pull request closed without merging**